simson/common/data_extrapolations.py

```diff
@@ -33,29 +26,42 @@ def regress(self):
         pass
 
 
+class OneDimensionalExtrapolation(Extrapolation):
+    @model_validator(mode="after")
+    def validate_data(self):
+        assert self.data_to_extrapolate.ndim == 1, "Data to extrapolate must be 1-dimensional."
+        assert self.target_range.ndim == 1, "Target range must be 1-dimensional."
+        assert self.data_to_extrapolate.shape[0] < self.target_range.shape[0], (
+            "data_to_extrapolate must be smaller then target_range")
```

Comment:
Last check should be moved to Extrapolation parent class. 
The function name in parent and subclass should then be different, such that inheriting just adds a second validator-decorated function.